### ISSUE TITLE
GDOs - An Idea

### ISSUE THREAD

**Issue by Zunth5** [PC-Logix/LanteaCraft#287]
An Idea;

-An item you can keep in inventory that when right clicked, will act like the GDOs in the series. A user would be able to enter a code of some length that the user defines. If there is a gate in proximity to the player that has an active connection, the receiving gate will return an event returning the code that the player input. 

This could be useful once there are iris so that IDCs can be used, but versatile enough to be used for other things. Correct me if I am wrong, but in the series it is implied on some occasions the GDOs have received signals indicating that the Stargate's iris had been opened. If this is true, then using CC; 

-A method could be called to have a dialed gate transmit a signal to all GDOs in range causing them to react in some way. Maybe turn an indication light on and off.

**Comment by Zunth5** [comment]
![gdo01](https://cloud.githubusercontent.com/assets/6885957/5825956/35897d48-a0bd-11e4-9881-1bc90a2ae6bc.jpg)

**Comment by Zunth5** [comment]
![stargate-atlantis-atlantis-shield-gdo-1](https://cloud.githubusercontent.com/assets/6885957/5826133/6de2bcc6-a0be-11e4-9c20-f2fc13eb442b.jpg)

**Comment by Kodos-Atoz** [comment]
Have you given any thought to just using an OpenComputers tablet?

**Comment by MichiyoRavencroft** [comment]
We are fully aware of what GDOs are, Lochie and myself are avid fans of the show, except for Universe :P  I can't speak for Lochie who is currently doing 99.5% of the work on the mod, but AFAIK something like them is planned.

**Comment by Zunth5** [comment]
I have not used OpenComputers in general, but find CC to be agreeable. While this is probably a fine alternative, the hope was more for something designed to work with the gates themselves on a practical level.

This idea is more of a mute point only because there are no Iris yet. My only thought is that while I don't particularly care if the mod is "conical" as per realism, I would rather have it be practical in terms of functionality (Like an adventure stepping through the gate). The reason I made suggestion now is because I believe it makes sense based on how this mod functions and that such a device would add to the experience once there are Iris.

I grew up with the show as it was on TV, and am only now going through season by season. I did not mean to make it sound as if you did not know what they were. I only wanted to be descriptive. :3 

-Heal-

**Comment by Kodos-Atoz** [comment]
ComputerCraft is a great tool for learning the basics of Lua on, but OpenComputers overall has entirely way more functionality and a tablet would be near perfect for what you're wanting a GDO for as far as I can tell.

**Comment by Zunth5** [comment]
I agree with you (based on your description) that OpenComputers could provide the functionality I describe. 

While this is one way to "shave" the metaphorical cat, it is not why this has been brought up. Relying on another mods devices for something that could be considered feature of another is a little absurd. Especially if the other is not a requirement of the first to use. For the same reason that modders in general have redesigned steps, doors, glass, and even energy sources; this device would be added because it would be tailored and made to be plug and play with lanteacraft. Although I do wonder if tweezers or shaving cream will be used....

And to be frank, I literally don't care if CC or OC is better, so long as I can vaporize endermen in my gateroom at the tap of a monitor.

-Heal-

**Comment by Sarous** [comment]
While Zunth is right in that needing other mods is painful- consider that GDOs don't work on regular stargates- indeed, the only gates in the series that even have Irises also have computer interfaces rather than a DHD. So, cannonly speaking, it isn't outragous to require a computer to make a GDO function.

**Comment by hehe3301** [comment]
Right now i have my own "version" of the GDO by sending a 16bit wireless redstone signal to extend a floor over my gateroom, without it if you go to my gate you get dropped into a lava pit

**Comment by Zunth5** [comment]
Sounds like a Goa'uld version of a iris; Cruel yet effective. I personally don't like a brisk swim in anything above 212°F.

Ill have to see what this wireless red stone signal is about. But I don't think it nullifies the original enhancement request.

**Comment by hehe3301** [comment]
It does not nullify the request, but this is a recommendation for use in the meantime

**Comment by AfterLifeLochie** [comment]
A dedicated GDO has been planned for a long time. I brushed on it in issue #124, but I don't have any estimate on when it'll be done par say.
